File: DaggerProject3/app/src/main/java/com/example/daggerproject3/model/Mobile.java
package com.example.daggerproject3.model;

import android.util.Log;

import javax.inject.Inject;

public class Mobile {

    private Battery battery;
    private Processor processor;
    private Camera camera;

    @Inject
    public Mobile(Battery battery, Processor processor,Camera camera) {
        this.battery = battery;
        this.processor = processor;
        this.camera=camera;
        Log.i("mobile", "Battery: "+battery.toString()+" "+this.toString());
    }

    public void run() {
        processor.start();
        Log.i("mobile", "mobile run: ");
    }

    @Inject
    public void setCharger(Charger charger) {
        Log.i("mobile", "setCharger: ");
    }
}

File: DaggerProject3/app/src/main/java/com/example/daggerproject3kotlin/model/mobile.kt
package com.example.daggerproject3kotlin.model

import android.util.Log
import com.example.daggerproject3kotlin.model.Battery
import javax.inject.Inject

class mobile @Inject constructor(battery: Battery, processor: Processor, camera: Camera) {

    init {
        Log.i("mobile", "mobile created: ")
    }

    fun run(){
        Log.i("mobile", "run: ")
    }
}